src/webview-ext.wkwebview.ts

```diff
@@ -144,3 +146,356 @@ export class WKScriptMessageHandlerImpl extends NSObject implements WKScriptMess
 }
+
+export class WKWebViewWrapper implements IOSWebViewWrapper {
+    protected wkWebViewConfiguration: WKWebViewConfiguration;
+    protected wkNavigationDelegate: WKNavigationDelegateImpl;
+    protected wkCustomUrlSchemeHandler: CustomUrlSchemeHandler;
+    protected wkUserContentController: WKUserContentController;
+    protected wkUserScriptInjectWebViewBrigde: Promise<WKUserScript> | void;
+    protected wkUserScriptViewPortCode: Promise<WKUserScript>;
+    protected wkNamedUserScripts = [] as Array<{
+        resourceName: string;
+        wkUserScript: WKUserScript;
+    }>;
+
+    public owner: WeakRef<WebViewExtBase>;
+    public get ios(): WKWebView | void {
+        const owner = this.owner.get();
+        return owner && owner.ios;
+    }
+
+    public readonly shouldInjectWebViewBridge = false;
+
+    public get autoInjectJSBridge() {
+        const owner = this.owner.get();
+        return owner && owner.autoInjectJSBridge;
+    }
+
+    constructor(owner: WebViewExtBase) {
+        this.owner = new WeakRef(owner);
+    }
+
+    public createNativeView() {
+        const owner = this.owner.get();
+        if (!owner) {
+            throw new Error("No owner, this should not happen");
+        }
+
+        const configuration = WKWebViewConfiguration.new();
+        this.wkWebViewConfiguration = configuration;
+
+        const messageHandler = WKScriptMessageHandlerImpl.initWithOwner(this.owner);
+        const wkUController = (this.wkUserContentController = WKUserContentController.new());
+        wkUController.addScriptMessageHandlerName(messageHandler, "nsBridge");
+        configuration.userContentController = wkUController;
+        configuration.preferences.setValueForKey(true, "allowFileAccessFromFileURLs");
+        configuration.setValueForKey(true, "allowUniversalAccessFromFileURLs");
+
+        this.wkCustomUrlSchemeHandler = new CustomUrlSchemeHandler();
+
+        configuration.setURLSchemeHandlerForURLScheme(this.wkCustomUrlSchemeHandler, owner.interceptScheme);
+
+        const webview = new WKWebView({
+            frame: CGRectZero,
+            configuration,
+        });
+
+        return webview;
+    }
+
+    public initNativeView() {
+        this.wkNavigationDelegate = WKNavigationDelegateImpl.initWithOwner(this.owner);
+
+        this.loadWKUserScripts();
+    }
+
+    public disposeNativeView() {
+        this.wkNavigationDelegate = null;
+        this.wkCustomUrlSchemeHandler = null;
+    }
+
+    public onLoaded() {
+        const ios = this.ios;
+        if (ios) {
+            ios.navigationDelegate = this.wkNavigationDelegate;
+        }
+    }
+
+    public onUnloaded() {
+        const ios = this.ios;
+        if (ios) {
+            ios.navigationDelegate = null;
+        }
+    }
+
+    public loadUrl(src: string) {
+        const owner = this.owner.get();
+        if (!owner) {
+            return;
+        }
+
+        const ios = this.ios;
+        if (!ios) {
+            return;
+        }
```

Comment:
A little verbose, perhaps just check `(!owner || !ios)` (repeated a couple of other places).